src/security_buffer.rs

```diff
@@ -0,0 +1,240 @@
+use std::fmt;
+use std::mem::take;
+
+use crate::{Error, ErrorKind, Result, SecurityBufferType};
+
+/// A special security buffer type is used for the data decryption. Basically, it's almost the same
+/// as [OwnedSecurityBuffer] but for decryption.
+///
+/// [DecryptMessage](https://learn.microsoft.com/en-us/windows/win32/secauthn/decryptmessage--general)
+/// "The encrypted message is decrypted in place, overwriting the original contents of its buffer."
+///
+/// So, the already defined [SecurityBuffer] is not suitable for decryption because it uses [Vec] inside.
```

Comment:
suggestion: I understand we should modify this docstring
```suggestion
/// So, the [OwnedSecurityBuffer] is not suitable for decryption because it uses [Vec] inside.
```

Reply:
Oh, I thought the IDE had updated them all. I'll review them